From e8b3a0e8740f51e7e942cbe0734dfe3b09c0b7d6 Mon Sep 17 00:00:00 2001
From: Andy CrossGate Yan <GeForce8800Ultra@gmail.com>
Date: Wed, 19 Jan 2022 18:04:36 +0000
Subject: [PATCH] DeskClock: Adapt app and daydream clocks to keyguard style

Ignore the configurable widget for now, as I don't use it anyway

Change-Id: Id9ab04d48b7cd67f7a2925db9b8d4e1f4cf6506c
---
 res/layout/date_and_next_alarm_time.xml       | 53 ++++++++++---------
 res/layout/desk_clock_saver.xml               | 10 +---
 res/layout/main_clock_frame.xml               | 36 ++++---------
 res/values/dimens.xml                         | 12 ++++-
 res/values/donottranslate.xml                 |  2 +-
 res/values/styles.xml                         | 18 +++++++
 res/xml/screensaver_settings.xml              | 16 ------
 src/com/android/deskclock/AlarmUtils.java     |  2 +-
 src/com/android/deskclock/ClockFragment.java  |  3 --
 src/com/android/deskclock/Utils.java          | 35 ++----------
 .../android/deskclock/data/SettingsDAO.java   |  2 +-
 11 files changed, 76 insertions(+), 113 deletions(-)

diff --git a/res/layout/date_and_next_alarm_time.xml b/res/layout/date_and_next_alarm_time.xml
index 9a0cb9103..b29ffedcf 100644
--- a/res/layout/date_and_next_alarm_time.xml
+++ b/res/layout/date_and_next_alarm_time.xml
@@ -18,36 +18,41 @@
     android:layout_width="wrap_content"
     android:layout_height="wrap_content"
     xmlns:tools="http://schemas.android.com/tools"
-    android:layout_gravity="center_vertical"
-    android:gravity="center">
+    android:gravity="center_vertical|start"
+    android:orientation="vertical">
 
     <TextView
         android:id="@+id/date"
-        style="@style/body"
+        style="@style/sc_keyguard_row"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:textAllCaps="true"
+        android:paddingTop="@dimen/sc_keyguard_status_area_top_padding"
+        android:paddingStart="@dimen/sc_keyguard_row_date_start_padding"
         tools:text="Sa., 01.01."/>
 
-    <TextView
-        android:id="@+id/nextAlarmIcon"
-        style="@style/body"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_gravity="center"
-        android:layout_marginEnd="@dimen/alarm_icon_padding"
-        android:layout_marginStart="@dimen/alarm_icon_padding"
-        android:ellipsize="none"
-        android:includeFontPadding="false"
-        android:maxLines="1"
-        android:text="@string/clock_emoji" />
-
-    <TextView
-        android:id="@+id/nextAlarm"
-        style="@style/body"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:textAllCaps="true"
-        tools:text="Mo., 07:00"/>
+    <LinearLayout
+         android:layout_width="wrap_content"
+         android:layout_height="wrap_content"
+         android:paddingTop="@dimen/sc_keyguard_row_top_padding"
+         android:paddingStart="@dimen/sc_keyguard_row_alarm_start_padding">
+
+        <TextView
+            android:id="@+id/nextAlarmIcon"
+            style="@style/sc_keyguard_row"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center"
+            android:layout_marginEnd="@dimen/alarm_icon_padding"
+            android:text="@string/clock_emoji"
+            android:textSize="@dimen/sc_keyguard_alarm_icon_size" />
+
+        <TextView
+            android:id="@+id/nextAlarm"
+            style="@style/sc_keyguard_row"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            tools:text="Mo., 07:00"/>
+
+    </LinearLayout>
 
 </LinearLayout>
diff --git a/res/layout/desk_clock_saver.xml b/res/layout/desk_clock_saver.xml
index c147bf7cd..d24219dbb 100644
--- a/res/layout/desk_clock_saver.xml
+++ b/res/layout/desk_clock_saver.xml
@@ -42,16 +42,10 @@
 
         <TextClock
             android:id="@+id/digital_clock"
-            style="@style/big_thin"
+            style="@style/sc_keyguard_clock"
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
-            android:layout_gravity="center"
-            android:layout_marginBottom="@dimen/bottom_text_spacing_digital"
-            android:ellipsize="none"
-            android:gravity="center"
-            android:singleLine="true"
-            android:textColor="@color/white"
-            android:textSize="@dimen/main_clock_font_size" />
+            android:fontFamily="sans-serif-thin" />
 
         <include layout="@layout/date_and_next_alarm_time" />
 
diff --git a/res/layout/main_clock_frame.xml b/res/layout/main_clock_frame.xml
index c26f61dbd..c2e84eaa3 100644
--- a/res/layout/main_clock_frame.xml
+++ b/res/layout/main_clock_frame.xml
@@ -26,44 +26,28 @@
     android:layout_marginEnd="24dp"
     android:orientation="vertical">
 
-    <androidx.constraintlayout.widget.ConstraintLayout
-        android:layout_width="match_parent"
+    <LinearLayout
+        android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_gravity="start">
+        android:gravity="center_vertical|start"
+        android:orientation="vertical">
 
         <com.android.deskclock.AnalogClock
             android:id="@+id/analog_clock"
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
-            android:layout_marginTop="@dimen/circle_margin_top"
-            app:layout_constraintBottom_toBottomOf="parent"
-            app:layout_constraintDimensionRatio="1:1"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toTopOf="parent"
-            app:layout_constraintWidth_percent="@dimen/analog_clock_width_percent"/>
+            android:layout_marginTop="@dimen/circle_margin_top"/>
 
         <com.android.deskclock.widget.AutoSizingTextClock
             android:id="@+id/digital_clock"
-            style="@style/display_time"
-            android:layout_width="match_parent"
+            style="@style/sc_keyguard_clock"
+            android:layout_width="wrap_content"
             android:layout_height="wrap_content"
-            android:ellipsize="none"
-            android:includeFontPadding="false"
             android:paddingTop="@dimen/main_clock_digital_padding"
-            android:singleLine="true"
-            android:textSize="@dimen/main_clock_digital_font_size"
-            app:layout_constraintBottom_toBottomOf="parent"
-            app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toTopOf="parent"
             tools:text="01:23"/>
 
-    </androidx.constraintlayout.widget.ConstraintLayout>
+        <include layout="@layout/date_and_next_alarm_time" />
+
+    </LinearLayout>
 
-    <include
-        layout="@layout/date_and_next_alarm_time"
-        android:id="@+id/date_and_next_alarm_time"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_gravity="start"/>
 </LinearLayout>
diff --git a/res/values/dimens.xml b/res/values/dimens.xml
index bf5128d1a..546cc02a8 100644
--- a/res/values/dimens.xml
+++ b/res/values/dimens.xml
@@ -60,7 +60,7 @@
 
     <dimen name="body_font_padding">4dp</dimen>
 
-    <dimen name="alarm_icon_padding">6dp</dimen>
+    <dimen name="alarm_icon_padding">7dp</dimen>
 
     <dimen name="no_alarms_size">90dp</dimen>
     <dimen name="fab_alarm_size">56dp</dimen>
@@ -129,4 +129,14 @@
     <dimen name="widget_background_corner_radius">80dp</dimen>
     <dimen name="widget_configuration_background_corner_radius">12dp</dimen>
     <dimen name="widget_padding">16dp</dimen>
+
+    <!-- Keyguard dimens, taken from S fwb -->
+    <dimen name="sc_keyguard_clock_text_size">86dp</dimen>
+    <dimen name="sc_keyguard_row_text_size">16dp</dimen>
+    <dimen name="sc_keyguard_alarm_icon_size">21dp</dimen>
+    <!-- Padding correction values, measured from S fwb -->
+    <dimen name="sc_keyguard_status_area_top_padding">8dp</dimen>
+    <dimen name="sc_keyguard_row_top_padding">11dp</dimen>
+    <dimen name="sc_keyguard_row_date_start_padding">6dp</dimen>
+    <dimen name="sc_keyguard_row_alarm_start_padding">5.5dp</dimen>
 </resources>
diff --git a/res/values/donottranslate.xml b/res/values/donottranslate.xml
index 1cbd61deb..918b5371e 100644
--- a/res/values/donottranslate.xml
+++ b/res/values/donottranslate.xml
@@ -34,7 +34,7 @@
     <!-- Default screensaver bold text. -->
     <bool name="default_clock_bold_text">false</bool>
     <!-- Default screensaver show AM/PM. -->
-    <bool name="default_clock_show_ampm">true</bool>
+    <bool name="default_clock_show_ampm">false</bool>
     <!-- Default screensaver night mode follows do no disturbe. -->
     <bool name="default_clock_night_mode_dnd">false</bool>
     <!-- Default screensaver night mode brightness. -->
diff --git a/res/values/styles.xml b/res/values/styles.xml
index 3cb625ba6..b46916317 100644
--- a/res/values/styles.xml
+++ b/res/values/styles.xml
@@ -304,4 +304,22 @@
         <item name="appWidgetThemeTransparent">@style/Theme.DeskClock.DigitalWidgetTransparentLightBg</item>
         <item name="appWidgetThemeSolid">@style/Theme.DeskClock.DigitalWidgetLightBg</item>
     </style>
+
+    <style name="sc_keyguard_clock">
+        <item name="android:fontFamily">sans-serif-light</item>
+        <item name="android:fontFeatureSettings">tnum</item>
+        <item name="android:textSize">@dimen/sc_keyguard_clock_text_size</item>
+        <item name="android:textColor">@color/white</item>
+        <item name="android:ellipsize">none</item>
+        <item name="android:includeFontPadding">false</item>
+        <item name="android:maxLines">1</item>
+    </style>
+
+    <style name="sc_keyguard_row">
+        <item name="android:textSize">@dimen/sc_keyguard_row_text_size</item>
+        <item name="android:textColor">@color/white</item>
+        <item name="android:ellipsize">none</item>
+        <item name="android:includeFontPadding">false</item>
+        <item name="android:maxLines">1</item>
+    </style>
 </resources>
diff --git a/res/xml/screensaver_settings.xml b/res/xml/screensaver_settings.xml
index 754c6980b..f57d3beb8 100644
--- a/res/xml/screensaver_settings.xml
+++ b/res/xml/screensaver_settings.xml
@@ -35,22 +35,6 @@
         android:title="@string/clock_color_title"
         app:iconSpaceReserved="false" />
 
-    <SwitchPreferenceCompat
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:defaultValue="@bool/default_clock_bold_text"
-        android:key="screensaver_bold_text"
-        android:title="@string/bold_text_title"
-        app:iconSpaceReserved="false" />
-
-    <SwitchPreferenceCompat
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:defaultValue="@bool/default_clock_show_ampm"
-        android:key="screensaver_show_ampm"
-        android:title="@string/show_ampm_title"
-        app:iconSpaceReserved="false" />
-
     <PreferenceCategory
         android:title="@string/night_mode_title"
         app:iconSpaceReserved="false" >
diff --git a/src/com/android/deskclock/AlarmUtils.java b/src/com/android/deskclock/AlarmUtils.java
index c3739bac8..5b931a46d 100644
--- a/src/com/android/deskclock/AlarmUtils.java
+++ b/src/com/android/deskclock/AlarmUtils.java
@@ -38,7 +38,7 @@ import java.util.Locale;
 public class AlarmUtils {
 
     public static String getFormattedTime(Context context, Calendar time) {
-        final String skeleton = DateFormat.is24HourFormat(context) ? "EHm" : "Ehma";
+        final String skeleton = DateFormat.is24HourFormat(context) ? "Hm" : "hma";
         final String pattern = DateFormat.getBestDateTimePattern(Locale.getDefault(), skeleton);
         return (String) DateFormat.format(pattern, time);
     }
diff --git a/src/com/android/deskclock/ClockFragment.java b/src/com/android/deskclock/ClockFragment.java
index 7591a18d0..8b0eecf7e 100644
--- a/src/com/android/deskclock/ClockFragment.java
+++ b/src/com/android/deskclock/ClockFragment.java
@@ -122,7 +122,6 @@ public final class ClockFragment extends DeskClockFragment {
             Utils.updateDate(mDateFormat, mDateFormatForAccessibility, mClockFrame);
             Utils.setClockStyle(mDigitalClock, mAnalogClock);
             Utils.setClockSecondsEnabled(mDigitalClock, mAnalogClock);
-            Utils.updateDateGravity(mClockFrame);
         }
 
         // Schedule a runnable to update the date every quarter hour.
@@ -150,7 +149,6 @@ public final class ClockFragment extends DeskClockFragment {
         if (mDigitalClock != null && mAnalogClock != null) {
             Utils.setClockStyle(mDigitalClock, mAnalogClock);
             Utils.setClockSecondsEnabled(mDigitalClock, mAnalogClock);
-            Utils.updateDateGravity(mClockFrame);
         }
 
         final View view = getView();
@@ -492,7 +490,6 @@ public final class ClockFragment extends DeskClockFragment {
                 Utils.updateDate(dateFormat, dateFormatForAccessibility, itemView);
                 Utils.setClockStyle(mDigitalClock, mAnalogClock);
                 Utils.setClockSecondsEnabled(mDigitalClock, mAnalogClock);
-                Utils.updateDateGravity(itemView);
             }
         }
     }
diff --git a/src/com/android/deskclock/Utils.java b/src/com/android/deskclock/Utils.java
index 27109a9d2..575f1fde4 100644
--- a/src/com/android/deskclock/Utils.java
+++ b/src/com/android/deskclock/Utils.java
@@ -50,9 +50,7 @@ import android.text.style.RelativeSizeSpan;
 import android.text.style.StyleSpan;
 import android.text.style.TypefaceSpan;
 import android.util.ArraySet;
-import android.view.Gravity;
 import android.view.View;
-import android.widget.LinearLayout;
 import android.widget.TextClock;
 import android.widget.TextView;
 
@@ -334,23 +332,6 @@ public class Utils {
         dateDisplay.setContentDescription(new SimpleDateFormat(descriptionPattern, l).format(now));
     }
 
-    public static void updateDateGravity(View clockFrame) {
-        View dateAndNextAlarm = clockFrame.findViewById(R.id.date_and_next_alarm_time);
-        LinearLayout.LayoutParams lp =
-                (LinearLayout.LayoutParams)dateAndNextAlarm.getLayoutParams();
-
-        final DataModel.ClockStyle clockStyle = DataModel.getDataModel().getClockStyle();
-        switch (clockStyle) {
-            case ANALOG:
-                lp.gravity = Gravity.CENTER;
-                break;
-            case DIGITAL:
-                lp.gravity = Gravity.START;
-                break;
-        }
-        dateAndNextAlarm.setLayoutParams(lp);
-    }
-
     /***
      * Formats the time in the TextClock according to the Locale with a special
      * formatting treatment for the am/pm label.
@@ -361,8 +342,7 @@ public class Utils {
     public static void setTimeFormat(TextClock clock, boolean includeSeconds) {
         if (clock != null) {
             // Get the best format for 12 hours mode according to the locale
-            clock.setFormat12Hour(get12ModeFormat(0.4f /* amPmRatio */, includeSeconds, true,
-                    true));
+            clock.setFormat12Hour(get12ModeFormat(0.4f /* amPmRatio */, includeSeconds));
             // Get the best format for 24 hours mode according to the locale
             clock.setFormat24Hour(get24ModeFormat(includeSeconds));
         }
@@ -377,17 +357,8 @@ public class Utils {
      */
     public static void setScreensaverTimeFormat(TextClock clock, boolean includeSeconds) {
         if (clock != null) {
-            final Boolean boldText = DataModel.getDataModel().getScreensaverBoldTextOn();
-            final Boolean showAmPm = DataModel.getDataModel().getScreensaverShowAmPmOn();
             // Get the best format for 12 hours mode according to the locale
-            CharSequence pattern = get12ModeFormat(0.4f /* amPmRatio */, includeSeconds, boldText,
-                    showAmPm);
-            final Spannable sp = new SpannableString(pattern);
-            if (boldText) {
-                sp.setSpan(new StyleSpan(Typeface.BOLD), 0, pattern.length(),
-                    Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
-            }
-            clock.setFormat12Hour(sp);
+            clock.setFormat12Hour(get12ModeFormat(0.4f /* amPmRatio */, includeSeconds));
             // Get the best format for 24 hours mode according to the locale
             clock.setFormat24Hour(get24ModeFormat(includeSeconds));
         }
@@ -400,7 +371,7 @@ public class Utils {
      * @return format string for 12 hours mode time, not including seconds
      */
     public static CharSequence get12ModeFormat(float amPmRatio, boolean includeSeconds) {
-        return get12ModeFormat(amPmRatio, includeSeconds, true, true);
+        return get12ModeFormat(amPmRatio, includeSeconds, true, false);
     }
 
     /**
diff --git a/src/com/android/deskclock/data/SettingsDAO.java b/src/com/android/deskclock/data/SettingsDAO.java
index f55ede8e6..1fa303d6e 100644
--- a/src/com/android/deskclock/data/SettingsDAO.java
+++ b/src/com/android/deskclock/data/SettingsDAO.java
@@ -211,7 +211,7 @@ final class SettingsDAO {
      * @return {@code true} if the screen saver should show AM/PM in 12 hour mode
      */
     static boolean getScreensaverShowAmPmOn(SharedPreferences prefs) {
-        return prefs.getBoolean(ScreensaverSettingsActivity.KEY_SHOW_AMPM, true);
+        return prefs.getBoolean(ScreensaverSettingsActivity.KEY_SHOW_AMPM, false);
     }
 
     /**
-- 
2.34.1

